### PR TITLE
Turn on strict mode and opt into VB 15.3

### PR DESCRIPTION
**Pull request by davkean**
- Turn on strict mode (https://github.com/dotnet/roslyn/search?q=FeatureStrictEnabled&unscoped_q=FeatureStrictEnabled)
- Opt into VB 15.3

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -12,13 +12,18 @@
...
   <PropertyGroup Condition="'$(Language)' == 'C#'">
     <CheckForOverflowUnderflow>false</CheckForOverflowUnderflow>
+    <!-- Build machines only have Visual Studio 15.5 -->
     <LangVersion>7.2</LangVersion>
+    <Features>strict</Features>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Language)' == 'VB'">
+    <!-- Build machines only have Visual Studio 15.5 (which comes with VB 15.3) -->
+    <LangVersion>15.3</LangVersion>
```
Comment (jmarolf) [inline src/Directory.Build.props]: We should be building with a compiler version from nuget not VS. Do 7.3 and 15.5 give errors?

Comment (Pilchie) [reply src/Directory.Build.props]: When I looked at it before, we are for the main build, but not for the xaml compiler build in markuppass1 :(

Comment (davkean) [reply src/Directory.Build.props]: Sounds like a bug? Is there one on this?

Comment (Pilchie) [reply src/Directory.Build.props]: Not sure?  @tmat?

Comment (tmat) [reply src/Directory.Build.props]: Could be that `Language` is not set properly in tmp_proj. .NET 4.7.2 changed the name of the temp project, so our detection of the temp proj in the toolset stopped working. I fixed that in recent version of the toolset.

**Comment by paul1956** [comment]
The current version of VS 2017 comes with VB 15.5 but by "default" exposes VB 15.0 unless you opt-in to something else. Why VB 15.3 instead of 15.5? It would be really nice to be able give parameter names to True and False. I have brought this up elsewhere but it seems to apply here as well.

**Comment by paul1956** [comment]
For your repo why not 15_5 because would be nice to use new feature when I add support for comments after _.

**Pull request merged**